blog/_posts/2016-08-22-GSoC2016-Graft.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: "GSoC 2016 Project: Graft.jl"
+author: <a href="https://github.com/pranavtbhat">Pranav Thulasiram Bhat</a>
+---
+
+This blog post describes my work on [Graft.jl](https://github.com/pranavtbhat/Graft.jl), a general purpose graph analysis package for Julia. For those unfamiliar with graphs, or the need to analyze them, this would be a good [readup](https://www.cl.cam.ac.uk/teaching/1011/PrincComm/slides/graph_theory_1-11.pdf).
+
+# Proposal
+My proposal, titled [ParallelGraphs](https://github.com/pranavtbhat/Gsoc2016/blob/master/Proposal.md), was to develop a parallelized/distributed graph algorithms
+library. However, in the first month or so, we decided to work towards a more general framework that supports data analysis on
+networks (graphs with attributes defined on vertices and edges).
+Our change in direction was mainly motivated by:
+
+- The challenges associated with distributed graph computations. [This](http://www.frankmcsherry.org/graph/scalability/cost/2015/01/15/COST.html)
+blog post was an eye opener.
+- Only super-massive graphs, of the order of terabytes or petabytes, require distributed execution. Most useful graphs can be analyzed on a single compute node.
+- Multi-threading is not yet fully established in Julia.
+- [LightGraphs.jl](https://github.com/JuliaGraphs/LightGraphs.jl) already provides fast implementations for most graph algorithms.
+
```

Comment:
As we looked at public datasets, we felt that the ability to combine graph theoretic analyses with real world data was the missing piece in Julia.